fhir-bulkimportexport/src/main/java/com/ibm/watson/health/fhir/bulkexport/BulkExportBatchLet.java

```diff
@@ -149,250 +160,230 @@
+            } catch (Exception e) {
+                logger.warning("process: Set page size to default(" + Constants.DEFAULT_SEARCH_PAGE_SIZE + ").");
+            }
+        }
+        if (!isSingleCosObject && cosBucketMaxFileSize != null) {
+            try {
+                cosMaxFileSize = Integer.parseInt(cosBucketMaxFileSize);
+                logger.info("process: Set max COS file size to " + cosMaxFileSize + ".");
+            } catch (Exception e) {
+                cosMaxFileSize = Constants.DEFAULT_MAXCOSFILE_SIZE;
+                logger.warning("process: Set max COS file size to default(" + Constants.DEFAULT_MAXCOSFILE_SIZE + ").");
+            }
+        }
+
         FHIRConfiguration.setConfigHome("./");
         FHIRRequestContext.set(new FHIRRequestContext(fhirTenant, fhirDatastoreId));
-
         FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper();
         FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
-
         Class<? extends Resource> resourceType = (Class<? extends Resource>) ModelSupport
                 .getResourceType(fhirResourceType);
         FHIRSearchContext searchContext;
         FHIRPersistenceContext persistenceContext;
         Map<String, List<String>> queryParameters = new HashMap<>();
-        String queryString;
+        String queryString = "&_sort=_lastUpdated";
 
-        queryString = "&_lastUpdated=ge" + fhirSearchFromDate + "&_lastUpdated=lt" + fhirSearchToDate
-                + "&_sort=_lastUpdated";
-        queryParameters.put("_lastUpdated", Collections.singletonList("ge" + fhirSearchFromDate));
-        queryParameters.put("_lastUpdated", Collections.singletonList("lt" + fhirSearchToDate));
-        queryParameters.put("_sort", Arrays.asList(new String[] { "_lastUpdated" }));
+        if (fhirSearchFromDate != null) {
+            queryString += ("&_lastUpdated=ge" + fhirSearchFromDate);
+            queryParameters.put("_lastUpdated", Collections.singletonList("ge" + fhirSearchFromDate));
+        }
+        if (fhirSearchToDate != null) {
+            queryString += ("&_lastUpdated=lt" + fhirSearchToDate);
+            queryParameters.put("_lastUpdated", Collections.singletonList("lt" + fhirSearchToDate));
+        }
 
+        queryParameters.put("_sort", Arrays.asList(new String[] { "_lastUpdated" }));
         searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters, queryString);
-
-        int pageNum = 1, exported = 0;
-
+        int pageNum = 1, exported = 0, totalExported = 0;
         searchContext.setPageSize(pageSize);
-        
-        String combinedJsons = null;
-        
+
         if (isSingleCosObject) {
-            combinedJsons = "[" + "\r\n";
             uploadId = COSUtils.startPartUpload(cosClient, cosBucketName, cosBucketObjectName);
         }
 
+        ByteArrayOutputStream bufferStream = new ByteArrayOutputStream();
```

Comment:
try(ByteArrayOutputStream bufferStream = new ByteArrayOutputStream();) 
as it is technically closeable.